stellargraph/data/explorer.py

```diff
@@ -796,6 +796,29 @@ class TemporalRandomWalk(GraphWalk):
     for the algorithm, only the relative differences (e.g. seconds, days, etc).
     """
 
+    def __init__(self, graph, graph_schema=None, seed=None):
+        super().__init__(graph, graph_schema=graph_schema, seed=seed)
+        self._edges, self._times = self.graph.edges(include_edge_weight=True)
+        self._validate_times()
+
+    def _validate_times(self):
+        (neg_time_locs,) = np.where(self._times < 0)
+        num_neg_times = len(neg_time_locs)
+        if num_neg_times > 0:
+            max_edges_shown = 10
+            neg_time_edges_formatted = ", ".join(
+                [
+                    str((self._edges[loc], self._times[loc]))
+                    for loc in neg_time_locs[:max_edges_shown]
+                ]
+            )
+            if num_neg_times > max_edges_shown:
+                neg_time_edges_formatted += " ..."
+            raise ValueError(
+                f"All edge times must be non-negative. Found {num_neg_times} negatives: "
```

Comment:
This message is good, but it doesn't quite match the style we're kinda sorta using elsewhere (at least, I and sometimes @kieranricardo do), maybe it could be:

```
graph: expected all edge types to be non-negative, found ...
```

In particular this tells the user which parameter the problem is with. Although maybe that's obvious in this case and so is pointless?

Reply:
Ah yeah that style was what I was going for, so I'll switch to that